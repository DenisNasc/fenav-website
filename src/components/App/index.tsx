import React from 'react';
import {makeStyles, createStyles, Theme} from '@material-ui/core/styles';
import {Grid} from '@material-ui/core';

import Header from '../Header';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    gridContainer: {
      height: '100vh',
    },
    gridHeader: {
      height: '90px',
    },
    gridBody: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: 'calc(100% - 90px - 60px)',
      border: '1px solid black',
    },
    gridFooter: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: '60px',
      border: '1px solid red',
    },
  })
);

const App = () => {
  const classes = useStyles();

  return (
    <Grid container spacing={2} className={classes.gridContainer}>
      <Grid item xs={12} className={classes.gridHeader}>
        <Header />
      </Grid>
      <Grid item xs={12} className={classes.gridBody}>
        <div data-testid="App">CORPO DA P√ÅGINA</div>
      </Grid>
      <Grid item xs={12} className={classes.gridFooter}>
        <div data-testid="App">FOOTER</div>
      </Grid>
    </Grid>
  );
};

export default App;
