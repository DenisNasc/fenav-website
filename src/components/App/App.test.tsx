import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '.';

describe('App Component', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  test('should match to the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
