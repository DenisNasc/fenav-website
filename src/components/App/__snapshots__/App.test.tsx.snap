// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match to the snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-gridContainer-1 MuiGrid-container MuiGrid-spacing-xs-2"
>
  <div
    className="MuiGrid-root makeStyles-gridHeader-2 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-header-108 MuiPaper-elevation4"
      data-testid="Header"
    >
      <div
        className="MuiContainer-root makeStyles-container-109 MuiContainer-maxWidthLg"
      >
        <svg
          className="makeStyles-logo-111"
        >
          FENAV.svg
        </svg>
        <p
          className="MuiTypography-root makeStyles-title-110 MuiTypography-body1"
          data-testid="header-title"
        >
          Faculdade de Engenharia Naval
        </p>
      </div>
      <div
        className="MuiTabs-root makeStyles-categoriesContainer-112"
      >
        <div
          className="MuiTabs-scroller MuiTabs-fixed"
          onScroll={[Function]}
          style={
            Object {
              "marginBottom": null,
              "overflow": "hidden",
            }
          }
        >
          <div
            aria-label="simple tabs example"
            className="MuiTabs-flexContainer"
            onKeyDown={[Function]}
            role="tablist"
          >
            <button
              aria-controls="simple-ItemHeader-0"
              aria-selected={true}
              className="MuiButtonBase-root MuiTab-root makeStyles-categorie-113 MuiTab-textColorInherit Mui-selected"
              data-testid="categorie-1"
              disabled={false}
              id="simple-tab-0"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                A Faculdade
              </span>
              <span
                className="PrivateTabIndicator-root-215 PrivateTabIndicator-colorSecondary-217 MuiTabs-indicator"
                style={Object {}}
              />
            </button>
            <button
              aria-controls="simple-ItemHeader-1"
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-categorie-113 MuiTab-textColorInherit"
              data-testid="categorie-2"
              disabled={false}
              id="simple-tab-1"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Como Ingressar
              </span>
            </button>
            <button
              aria-controls="simple-ItemHeader-2"
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-categorie-113 MuiTab-textColorInherit"
              data-testid="categorie-3"
              disabled={false}
              id="simple-tab-2"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Notícias
              </span>
            </button>
            <button
              aria-controls="simple-ItemHeader-3"
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-categorie-113 MuiTab-textColorInherit"
              data-testid="categorie-4"
              disabled={false}
              id="simple-tab-3"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Publicações
              </span>
            </button>
            <button
              aria-controls="simple-ItemHeader-4"
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-categorie-113 MuiTab-textColorInherit"
              data-testid="categorie-5"
              disabled={false}
              id="simple-tab-4"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Pós-Graduação
              </span>
            </button>
            <button
              aria-controls="simple-ItemHeader-5"
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root makeStyles-categorie-113 MuiTab-textColorInherit"
              data-testid="categorie-6"
              disabled={false}
              id="simple-tab-5"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Área do Aluno
              </span>
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-root makeStyles-gridBody-3 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      data-testid="App"
    >
      CORPO DA PÁGINA
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-gridFooter-4 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      data-testid="App"
    >
      FOOTER
    </div>
  </div>
</div>
`;
